packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,230 @@
+      return false;
+    } else {
+      const fieldsMatch = vary.split(/\s*,\s*/).every(field => {
+        return cached.reqHeaders.get(field) === req.headers.get(field);
+      });
+      if (!fieldsMatch) {
+        return false;
+      }
+    }
+  }
+  if (cached.integrity) {
+    return ssri.parse(cached.integrity).match(cached.cacheIntegrity);
+  }
+  reqUrl.hash = null;
+  cacheUrl.hash = null;
+  return url.format(reqUrl) === url.format(cacheUrl);
+};
+
+/**
+ * This is an implementation of the Cache standard for make-fetch-happen using
+ * the Keyv storage interface. The primary target is Redis.
+ * docs: https://developer.mozilla.org/en-US/docs/Web/API/Cache
+ * https://www.npmjs.com/package/keyv
+ */
+export class CacheManagerKeyv implements ICacheManager {
+  public keyv;
+
+  constructor(public connection?: string) {
+    if (connection) {
+      this.keyv = new Keyv(connection, { keepAlive: false });
+      this.keyv.on("error", console.error);
+    }
+  }
+
+  /**
+   * Returns a Promise that resolves to the response associated with the first
+   * matching request in the Cache object.
+   *
+   * @param req The request to check for a cached response
+   * @param opts Currently for compatibility
+   */
+  async match(req: fetch.Request, opts?: any): Promise<fetch.Response> {
+    const metadataKey: string = getMetadataKey(req);
+    const contentKey: string = getContentKey(req);
+
+    // No match if there's no metadata for this request
+    const redisInfo = await this.keyv.get(metadataKey);
+    if (!redisInfo) {
+      return;
+    }
+
+    // No match if the metadata doesn't match
+    if (
+      !matchDetails(req, {
+        url: redisInfo.metadata.url,
+        reqHeaders: new fetch.Headers(redisInfo.metadata.reqHeaders),
+        resHeaders: new fetch.Headers(redisInfo.metadata.resHeaders),
+        cacheIntegrity: redisInfo.integrity,
+        integrity: opts && opts.integrity
+      })
+    ) {
+      return;
+    }
+
+    // Add customer headers to the response that include caching info
+    const resHeaders: fetch.Headers = new fetch.Headers(
+      redisInfo.metadata.resHeaders
+    );
+    addCacheHeaders(
+      resHeaders,
+      "",
+      contentKey,
+      redisInfo.integrity,
+      redisInfo.time
+    );
+
+    // Return the response without a body for HEAD requests
+    if (req.method === "HEAD") {
+      return new fetch.Response(null, {
+        url: req.url,
+        headers: resHeaders,
+        status: 200
+      });
+    }
+
+    // Get the cached response body
+    const body: string = await this.keyv.get(getContentKey(req));
+
+    return Promise.resolve(
+      new fetch.Response(Buffer.from(body), {
+        url: req.url,
+        headers: resHeaders,
+        status: 200,
+        size: redisInfo.size
+      })
+    );
+  }
+
+  /**
+   * Takes both a request and its response and adds it to the given cache.
+   *
+   * @param req The request this is a storing a response for
```

Comment:
Change to ```@param req The request to be stored```

Reply:
We're not storing the request, we're storing the response